src/alchemlyb/tests/test_ti_gaussian_quadrature.py

```diff
@@ -0,0 +1,168 @@
+"""Tests for all TI-based estimators in ``alchemlyb``.
+
+"""
+import pandas as pd
+import numpy as np
+import pytest
+
+import alchemlyb
+from alchemlyb.estimators import TI_GQ
+from alchemlyb.parsing import amber
+
+
+@pytest.fixture
+def ethanol_Coulomb(gmx_ethanol_Coulomb_dHdl):
+    dHdl = alchemlyb.concat(gmx_ethanol_Coulomb_dHdl)
+    return dHdl
+
+
+@pytest.fixture
+def ethanol_VDW(gmx_ethanol_VDW_dHdl):
+    dHdl = alchemlyb.concat(gmx_ethanol_VDW_dHdl)
+    return dHdl
+
+
+@pytest.fixture
+def ethanol(ethanol_Coulomb, ethanol_VDW):
+    dHdl = alchemlyb.concat([ethanol_Coulomb, ethanol_VDW])
+    return dHdl
+
+
+@pytest.fixture
+def tyk2_complex(amber_tyk2_example_complex):
+    dHdl = alchemlyb.concat(amber_tyk2_example_complex)
+    return dHdl
+
+
+@pytest.fixture
+def benzene_VDW(gmx_benzene_VDW_dHdl):
+    dHdl = alchemlyb.concat(gmx_benzene_VDW_dHdl)
+    return dHdl
+
+
+class TIestimatorMixin:
+    def test_get_delta_f(self, X_delta_f):
+        dHdl, E, dE = X_delta_f
+        est = self.cls().fit(dHdl)
+        # Use .iloc[0, -1] as we want to cater for both
+        # delta_f_.loc[0.0, 1.0] and delta_f_.loc[(0.0, 0.0), (0.0, 1.0)]
+        delta_f = est.delta_f_.iloc[0, -1]
+        d_delta_f = est.d_delta_f_.iloc[0, -1]
+
+        assert E == pytest.approx(delta_f, rel=1e-3)
+        assert dE == pytest.approx(d_delta_f, rel=1e-3)
+
+
+class TestTIGQ(TIestimatorMixin):
+    """Tests for TI_GQ."""
+
+    cls = TI_GQ
+
+    T = 300.0
+    kT_amber = amber.k_b * T
+
+    @pytest.fixture(
+        params=[
+            ("ethanol_Coulomb", 10.597, 0.04498),
+            ("ethanol_VDW", -3.340, 0.07548),
+            ("ethanol", 7.257, 0.08786),
+            ("tyk2_complex", -50.504, 0.09930),
+        ],
+    )
+    def X_delta_f(self, request):
+        get_dHdl, E, dE = request.param
+        return request.getfixturevalue(get_dHdl), E, dE
+
+
+def test_TI_GQ_separate_dhdl_multiple_column(ethanol):
+    dHdl = ethanol
+    estimator = TI_GQ().fit(dHdl)
+    assert isinstance(estimator.separate_dhdl()[0], list)
```

Comment:
It is good to test this estimator.separate_dhdl() but might be more clear to make this into a class and test various aspects. In theory one test should only be testing one thing.